### PR TITLE
Enable code analysis on shim project

### PR DESCRIPTION
**Pull request by pakrym**

### CODE REVIEW

File: src/AspNetCoreModuleV2/CommonLib/ResultException.h [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#pragma once
```
Comment (jkotalik) [inline src/AspNetCoreModuleV2/CommonLib/ResultException.h]: header.

File: src/AspNetCoreModuleV2/CommonLib/HandleWrapper.h [hunk 2]
```diff
@@ -10,37 +10,41 @@ struct InvalidHandleTraits
...
     using HandleType = HMODULE;
-    static constexpr HMODULE DefaultHandle = NULL;
-    static void Close(HMODULE handle) { FreeModule(handle); }
+    static constexpr HMODULE DefaultHandle = nullptr;
+    static void Close(HMODULE handle) noexcept { FreeModule(handle); }
 };
 
+// Code analysis doesn't like nullptr usages via traits
+#pragma warning( push )
+#pragma warning ( disable : 26477 ) // disable  Use 'nullptr' rather than 0 or NULL (es.47).
```
Comment (jkotalik) [inline src/AspNetCoreModuleV2/CommonLib/HandleWrapper.h]: Can we just swap to NULL for these values?

File: build/Build.Settings [hunk 3]
```diff
@@ -20,6 +20,11 @@
    <PropertyGroup Condition="'$(Configuration)' == 'Release'">
      <UseDebugLibraries>false</UseDebugLibraries>
      <WholeProgramOptimization>true</WholeProgramOptimization>
+     <CodeAnalysisRuleSet>..\DefaultRules.ruleset</CodeAnalysisRuleSet>
+     <RunCodeAnalysis>false</RunCodeAnalysis>
+     <!-- Do not enable code analysis on old compiler versions -->
+     <RunCodeAnalysis Condition="'$(VCToolsRedistVersion)' != '' AND '$(VCToolsRedistVersion)' >= 14.15.26706">true</RunCodeAnalysis>
```
Comment (jkotalik) [inline build/Build.Settings]: The only build this failed on before was the release build: https://ci3.dot.net/job/aspnet_IISIntegration/job/release_2.2/job/windows-Configuration_Release_prtest/669/console. Maybe you missed a flag for release?

Comment (pakrym) [reply build/Build.Settings]: Which flag?

File: src/AspNetCoreModuleV2/CommonLib/HandleWrapper.h (hunk 2 above)
Comment (pakrym) [reply src/AspNetCoreModuleV2/CommonLib/HandleWrapper.h]: They were NULL and were changed to nullptr but it didn't help.

Comment (jkotalik) [reply src/AspNetCoreModuleV2/CommonLib/HandleWrapper.h]: I see.

File: build/Build.Settings (hunk 3 above)
Comment (jkotalik) [reply build/Build.Settings]: Nvrm, didn't realize the build isn't run for debug.

**Pull request merged**